docs/user-guide/configuring.md

```diff
@@ -713,6 +713,61 @@ module.exports = {
 }
 ```
 
+## Configuration Based on Glob Patterns
+
+Sometimes a more fine-controlled configuration is necessary, for example if the configuration for files within the same directory has to be different. Therefore you can provide configurations under the `overrides` key that will only apply to files that match specific glob patterns, using the same format you would pass on the command line (e.g., `app/**/*.test.js`).
+
+### How it works
+
+* Glob pattern overrides can only be configured within `.eslintrc` files.
+* The patterns are applied against the file path relative to the directory of the `.eslintrc` file. For example, if your `.eslintrc` file has the path `/Users/john/workspace/any-project/.eslintrc` and the file you want to lint has the path `/Users/john/workspace/any-project/lib/util.js`, then the pattern provided in `.eslintrc` will be executed against the relative path `lib/util.js`.
+* Glob pattern overrides have higher precedence than the regular configuration in the same `.eslintrc` file.
```

Comment:
Can you clarify the precedence between two overrides blocks in the same file? Does the later override take precedence?

```json
{
  "overrides": [
    {
      "files": ["foo.js"],
      "config": {
        "rules": {
          "semi": ["error", "always"]
        }
      }
    },
    {
      "files": ["*.js"],
      "config": {
        "rules": {
          "semi": ["error", "never"]
        }
      }
    }
  ]
}
```

More specifically, what would be the config for `foo.js` in the above example?